File: spring-java/src/main/java/com/jmpark/springjava/entity/TimeTableEntity.java
package com.jmpark.springjava.entity;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.IdClass;

@Entity
@RequiredArgsConstructor
@Getter
@ToString
@EqualsAndHashCode(callSuper = false)
@IdClass(AreaId.class)
public class TimeTableEntity extends BaseTimeEntity {
    @Id
    String areaId = "";
    @Id
    String type = "";

    String openAt = null;
    String closeAt = null;
}

File: spring-kotlin/src/main/kotlin/com/jmpark/springkotlin/entity/TimeTableEntity.kt
package com.jmpark.springkotlin.entity

import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.IdClass

@Entity
@IdClass(AreaId::class)
data class TimeTableEntity (
    @Id
    var areaId: String = "",
    @Id
    var type: String = "",

    var openAt: String? = null,
    var closeAt: String? = null
) : BaseTimeEntity()